client/post-editor/media-modal/markup.js

```diff
@@ -148,13 +148,19 @@ Markup = {
 				urlOptions = { size: options.size };
 			}
 
-			const img = React.createElement( 'img', {
+			const isTransient = ( 0 === ( media.URL || '' ).indexOf( 'blob:' ) ) || !! media.transient;
```

Comment:
Mmm... at what points would we reach here with a `blob:` and where `media.transient` isn't truthy? Should we always ensure that `media.transient` is set as `true` when a blob URL?